### PR TITLE
Task_DEV-3

### PR DESCRIPTION
**Pull request by MaryVazhnik**

### CODE REVIEW

File: DEV_3/CalculationMethod.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Text;
+using System.Collections;
+
+namespace DEV_3
+{
+        public class CalculationMethod
+        {
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: has to be const

File: DEV_3/CalculationMethod.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Text;
+using System.Collections;
+
+namespace DEV_3
+{
+        public class CalculationMethod
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: Bad class method, no comments

File: DEV_3/CalculationMethod.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            /// Translates from the decimal number system to the number system with the base of number system
+            /// </summary>
+            /// <param name="number">The number that we convert</param>
+            /// <param name="sys">Number system, into which we convert</param>
+            /// <returns></returns>
+            public string ToNumberSystem(string number, string sys)
+            {
+                string newNum = String.Empty;
+                int num = Convert.ToInt32(number);
+                int chast = Convert.ToInt32(number);
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: What is chast? Why you recalculate it?

File: DEV_3/CalculationMethod.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                ArrayList numTemp = new ArrayList();
+                while (chast > 0)
+                {
+                    chast = chast / Convert.ToInt32(sys);
+                    numTemp.Add(num - chast * Convert.ToInt32(sys));
+                    num = chast;
+                }
+                int j;
+                for (j = numTemp.Count - 1; j >= 0; j--)
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: Code style

File: DEV_3/CalculationMethod.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                Console.WriteLine(newNum);
+                return newNum;
+            }
+            /// <summary>
+            /// Function, replaces letters with numbers and vice versa
+            /// </summary>
+            /// <param name="worknumber">Number to work over</param>
+            /// <param name="partytoactionplace">In which direction is the action performed with respect to the decimal number system</param>
+            /// <returns>Returns a letter if the number matches a letter and vice versa, otherwise the number</returns>
+            string newCh(string worknumber, string partytoactionplace)
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: Code style

File: DEV_3/CalculationMethod.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            {
+                string line = String.Empty;
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: Use StringBuilder

File: DEV_3/CalculationMethod.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    if (Convert.ToInt32(worknumber) > 10)
+                    {
+                        line += letters.Substring(Convert.ToInt32(worknumber) - 10, 1);
+                    }
+                    else
+                    {
+                        line += worknumber;
+                    }
+                }
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: Unclear logics, bad names,

File: DEV_3/ConverterOfNumeralSystems.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+            int check = Int32.Parse(args[1]);
+            if (check <= 20 & check >= 2)
+            {
+                prog.ToNumberSystem(args[0], args[1]);
+            }
+            else
+            {
+                Console.WriteLine("Error! Change the data, please");
+            }
+
```
Comment (IgorKheidorov) [inline DEV_3/ConverterOfNumeralSystems.cs]: Code style

File: DEV_3/CalculationMethod.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            string letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+            /// <summary>
+            /// Translates from the decimal number system to the number system with the base of number system
+            /// </summary>
+            /// <param name="number">The number that we convert</param>
+            /// <param name="sys">Number system, into which we convert</param>
+            /// <returns></returns>
+            public string ToNumberSystem(string number, string sys)
+            {
+                string newNum = String.Empty;
```
Comment (IgorKheidorov) [inline DEV_3/CalculationMethod.cs]: Are you sure that number will be converted? Use Parse and catch exceptions

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: DEV_3/CalculationMethod.cs (hunk 3 above)
Comment (MaryVazhnik) [reply DEV_3/CalculationMethod.cs]: chast to find the remainder

File: DEV_3/CalculationMethod.cs (hunk 9 above)
Comment (MaryVazhnik) [reply DEV_3/CalculationMethod.cs]: But I'm doing an argument check in main.

**Review by AlexanderSoroka — APPROVED** [review]
Spaghetti code. Must be re-written from scratch

**Pull request merged**